src/collector/models.py

```diff
@@ -1,127 +1,71 @@
-		self.wall3 = transcription.wall3
-		self.wall4 = transcription.wall4
-		self.wall5 = transcription.wall5
-		self.wall6 = transcription.wall6
 
-		self.link1 = transcription.link1
-		self.link2 = transcription.link2
-		self.link3 = transcription.link3
-		self.link4 = transcription.link4
-		self.link5 = transcription.link5
-		self.link6 = transcription.link6
+	# Metadata
+	puzzle_piece = models.ForeignKey(PuzzlePiece, on_delete=models.CASCADE, related_name="transcriptions", null=True, default=None)
+	submitter = models.CharField(max_length=64, verbose_name="Hash of submitter ip address", default="", blank=True)
+	submission_date = models.DateTimeField(verbose_name="Submission date", auto_now_add=True)
+	# Flags
+	bad_flag = models.BooleanField(verbose_name="Image is bad or hard to read", default=False)
+	rotation_flag = models.BooleanField(verbose_name="Image is rotated", default="")
+	# Center in short notation
+	center = models.CharField(max_length=1, verbose_name="center", default="")
+	# Walls
+	wall1 = models.BooleanField(verbose_name="Wall 1 (top)", default=False)
+	wall2 = models.BooleanField(verbose_name="Wall 2 (top-right)", default=False)
+	wall3 = models.BooleanField(verbose_name="Wall 3 (bottom-right)", default=False)
+	wall4 = models.BooleanField(verbose_name="Wall 4 (bottom)", default=False)
+	wall5 = models.BooleanField(verbose_name="Wall 5 (bottom-left)", default=False)
+	wall6 = models.BooleanField(verbose_name="Wall 6 (top-left)", default=False)
+	# Links
+	link1 = models.CharField(max_length=7, verbose_name="Link 1 (top)", default="")
+	link2 = models.CharField(max_length=7, verbose_name="Link 2 (top-right)", default="")
+	link3 = models.CharField(max_length=7, verbose_name="Link 3 (bottom-right)", default="")
+	link4 = models.CharField(max_length=7, verbose_name="Link 4 (bottom)", default="")
+	link5 = models.CharField(max_length=7, verbose_name="Link 5 (bottom-left)", default="")
+	link6 = models.CharField(max_length=7, verbose_name="Link 6 (top-left)", default="")
+	# SHA256 hash of 'center wallsAsBitstring link1 link2 link3 link4 link5 link6' for comparison
+	hash = models.CharField(max_length=64, verbose_name="SHA256 hash for comparison", default="", null=True, blank=True)
+
+	def save(self, *args, **kwargs):
+		self.sanitize_fields()
+		if self.bad_flag:
+			self.hash = None
+		else:
+			self.hash = self.calculate_hash()
+		super().save(*args, **kwargs)
+
+	def calculate_hash(self):
+		hashInput = (
+			f"{self.center} "
+			f"{str(int(self.wall1))}{str(int(self.wall2))}{str(int(self.wall3))}"
+			f"{str(int(self.wall4))}{str(int(self.wall5))}{str(int(self.wall6))} "
+			f"{self.link1} {self.link2} {self.link3} {self.link4} {self.link5} {self.link6}"
+		).encode("utf-8")
+		return sha256(hashInput).hexdigest()
+
+	def sanitize_fields(self):
+		fields = [ "center", "link1", "link2", "link3", "link4", "link5", "link6" ]
+		for field in fields:
+			self.__setattr__(field, self.__getattribute__(field).upper())
```

Comment:
I had a discussion with someone in #bug-reports that they found two transcriptions for separate images, with the same transcription, but different hashes.  

I was not able to reproduce the hashes (and the no hash bug was found at this time, and more pressing) but my only guess is that there were extra white spaces in the center and/or link fields. I see that you've limited the field sizes appropriately to 1/7, but I'd suggest append `.strip()` to the field sanitation to prevent rejecting possibly valid transcriptions but just with extra spaces.